condition/task.go

```diff
@@ -0,0 +1,181 @@
+package condition
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+var (
+	errInvalidTaskJSON = errors.New("invalid Task JSON")
+	errTaskUpdate      = errors.New("invalid Task update")
+)
+
+const (
+	TaskVersion1 = "1.0"
+	// Task KV bucket name
+	TaskKVRepositoryBucket = "tasks"
+)
+
+func TaskKVRepositoryKey(facilityCode string, conditionKind Kind, serverID string) string {
+	return fmt.Sprintf("%s.%s.%s", facilityCode, conditionKind, serverID)
+}
+
+type Task[P, D any] struct {
```

Comment:
I feel like using generics here is overkill if it's just basically going to be `any` when we deserialize the payload.

Reply:
At first sight it may seem like overkill, although, this enables controllers to define their own parameterized `Task` type based on top of this,

For examples, check out
- https://github.com/metal-toolbox/flasher/blob/36aaead5ce2103b0898ab9199bc8ff51f62a1795/internal/model/task.go#L52
- https://github.com/equinixmetal/metal-broker/blob/f019eb2f6445f7eeee9133db6f0bf65682e45ad9/internal/model/task.go#L13

That said, we can discuss alternative approaches that would fit better - and this can be done before flasher and broker goes to production. For now, I think its best we go with this as the first revision of the Task object so as to enable the inband work to come together quicker.